src/openforms/registrations/contrib/objects_api/api.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any
+
+from django.urls import path
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import authentication, permissions, serializers, views
+
+from openforms.api.views import ListMixin
+
+from .client import get_objecttypes_client
+
+
+class ObjecttypeSerializer(serializers.Serializer):
+    # Keys are defined in camel case as this is what we get from the Objecttype API
+    url = serializers.URLField(
+        label=_(
+            "URL reference to this object. This is the unique identification and location of this object."
+        ),
+    )
+    uuid = serializers.UUIDField(label=_("Unique identifier (UUID4)."))
+    name = serializers.CharField(label=_("Name of the object type."))
+    namePlural = serializers.CharField(label=_("Plural name of the object type."))
+
+
+class ObjecttypeVersionSerializer(serializers.Serializer):
+    version = serializers.IntegerField(
+        label=_("Integer version of the Objecttype."),
+    )
+
+
+@extend_schema_view(
+    get=extend_schema(
+        summary=_("List available Objecttypes"),
+        tags=["registration"],
+    ),
+)
+class ObjecttypesListView(ListMixin, views.APIView):
+    """
+    List the available Objecttypes.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = ObjecttypeSerializer
+
+    def get_objects(self) -> list[dict[str, Any]]:
+        with get_objecttypes_client() as client:
+            return client.list_objecttypes()
+
+
+class ObjecttypeVersionsListView(ListMixin, views.APIView):
+    """
+    List the available versions for an Objecttype.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = ObjecttypeVersionSerializer
+
+    def get_objects(self) -> list[dict[str, Any]]:
+        with get_objecttypes_client() as client:
+            return client.list_objecttype_versions(self.kwargs["submission_uuid"])
+
+
+app_name = "objects_api"
+
+urlpatterns = [
+    path(
+        "object-types",
+        ObjecttypesListView.as_view(),
+        name="object-types",
+    ),
+    path(
+        "object-types/<uuid:submission_uuid>/versions",
+        ObjecttypeVersionsListView.as_view(),
+        name="object-type-versions",
+    ),
+]
```

Comment:
Maybe split this file in `views.py`, `serializers.py` and `urls.py` so that it's more intuitive to find things?

Reply:
I've mirrored the `camunda` contrib module, but I can make the change to both modules if required